### PR TITLE
[freebox] Never log the application token

### PR DESCRIPTION
**Pull request by lolodomo**
Automatic update of the thing configuration with the application token

Console command to display the application token

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
Related to #3990

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/handler/FreeboxHandler.java [hunk 1]
```diff
@@ -293,6 +306,10 @@ public FreeboxApiManager getApiManager() {
         return apiManager;
     }
 
+    public String getAppToken() {
+        return getConfigAs(FreeboxServerConfiguration.class).appToken;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/handler/FreeboxHandler.java:310]: Can you use the cached config instead?
```suggestion
        return configuration.appToken;
```

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/console/FreeboxCommandExtension.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            }
+        } else {
+            printUsage(console);
+        }
+    }
+
+    @Override
+    public List<String> getUsages() {
+        return Arrays
+                .asList(new String[] { buildCommandUsage("<bridgeUID> " + APP_TOKEN, "show the application token") });
```
Comment (cpmeister) [inline bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/console/FreeboxCommandExtension.java:85]:
```suggestion
                .asList(buildCommandUsage("<bridgeUID> " + APP_TOKEN, "show the application token"));
```

Comment (lolodomo) [reply bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/console/FreeboxCommandExtension.java:85]: Fixed.

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/handler/FreeboxHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/handler/FreeboxHandler.java:310]: There was no global variable `configuration` but I added it.

File: bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/console/FreeboxCommandExtension.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    @Override
+    public void execute(String[] args, Console console) {
+        if (args.length == 2) {
+            FreeboxHandler handler = null;
+            try {
+                ThingUID thingUID = new ThingUID(args[0]);
+                Thing thing = thingRegistry.get(thingUID);
+                if ((thing != null) && (thing.getHandler() != null) && (thing.getHandler() instanceof FreeboxHandler)) {
+                    handler = (FreeboxHandler) thing.getHandler();
+                }
```
Comment (cpmeister) [inline bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/console/FreeboxCommandExtension.java:59]: This better deals with null checks.
```suggestion
                if(thing != null){
                    ThingHandler thingHandler = thing.getHandler();
                    if(thingHandler instanceof FreeboxHandler){
                        handler = (FreeboxHandler) thingHandler;
                    }
                }
```

Comment (lolodomo) [reply bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/console/FreeboxCommandExtension.java]: Done

**Pull request merged**